import-export-cli/cmd/mg/deleteAPI.go

```diff
@@ -0,0 +1,95 @@
+/*
+*  Copyright (c) WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+*
+*  WSO2 Inc. licenses this file to you under the Apache License,
+*  Version 2.0 (the "License"); you may not use this file except
+*  in compliance with the License.
+*  You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied.  See the License for the
+* specific language governing permissions and limitations
+* under the License.
+ */
+
+package mg
+
+import (
+	"encoding/base64"
+	"fmt"
+
+	"github.com/spf13/cobra"
+	mgImpl "github.com/wso2/product-apim-tooling/import-export-cli/impl/mg"
+	"github.com/wso2/product-apim-tooling/import-export-cli/utils"
+	"golang.org/x/crypto/ssh/terminal"
+)
+
+var (
+	deleteApisCmdAPIName    string
+	deleteApisCmdAPIVersion string
+	deleteApisCmdAPIVHost   string
+	deleteApisUsername      string
+)
+
+const deleteApisCmdLiteral = "apis"
+const deleteApisCmdShortDesc = "Delete API"
+const deleteApisCmdLongDesc = `Delete an API by specifying name, version and optionally vhost
+ by specifying the flags (--name (-n), --version (-v), and optionally --vhost (-vh)`
+
+var deleteApisCmdExamples = utils.ProjectName + ` ` + mgCmdLiteral + ` ` + deleteApisCmdLiteral + `-h https://localhost:9095 -u admin
+  ` + utils.ProjectName + ` ` + mgCmdLiteral + ` ` + deleteApisCmdLiteral + ` -n petstore -v 0.0.1 -vh www.pets.com http -h https://localhost:9095 -u admin
+  ` + utils.ProjectName + ` ` + mgCmdLiteral + ` ` + deleteApisCmdLiteral + ` -n "petstore VIP" -v 0.0.1 -h https://localhost:9095 -u admin`
+
+var mgDeleteAPIsResourcePath = "/apis/delete"
+
+// DeleteApisCmd represents the apis command
+var DeleteApisCmd = &cobra.Command{
+	Use:     deleteApisCmdLiteral,
+	Short:   deleteApisCmdShortDesc,
+	Long:    deleteApisCmdLongDesc,
+	Example: deleteApisCmdExamples,
+	Run: func(cmd *cobra.Command, args []string) {
+		utils.Logln(utils.LogPrefixInfo + deleteApisCmdLiteral + " called")
+
+		// handle auth
+		fmt.Print("Enter Password: ")
```

Comment:
I think we should support giving the password as a flag and if not provided only, it should prompt and ask. Please modify the code to handle that.

Also modify the descriptions,examples with the password change

Reply:
I did not include a password flag in order to avoid the password being saved in the terminal history. Do you think it is necessary to have the password flag?

Reply:
IMO it is good to have a flag for a password since these commands are used in the scripts when users execute CI/CD flows. So that scripts can be defined with the password and no human interaction needed when scripts are executing.

Reply:
I see. I'll add the password flag. Thanks. I hadn't thought about it.